### PR TITLE
test: Code Coverage: Camera Options Popup Component #1

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at f0c74f4</samp>

Refactored and tested the camera options popup component for uploading receipt images. Added type annotations, extracted a callback method, and used ElementRef in `camera-options-popup.component.ts`. Added a data-testid attribute in `camera-options-popup.component.html`. Created a new file `camera-options-popup.component.spec.ts` with unit tests for the component.

### <samp>🤖 Generated by Copilot at f0c74f4</samp>

> _`CameraOptionsPopup` is the gate to hell_
> _Upload your receipts or face the wrath_
> _Mock the dependencies, test the methods_
> _`ElementRef` will unleash the beast_

### <samp>🤖 Generated by Copilot at f0c74f4</samp>

*  Added a data-testid attribute to the camera-options--element div for testing purposes ([link](https://github.com/fylein/fyle-mobile-app/pull/2424/files?diff=unified&w=0#diff-683a514ef8854236cf9cc47d3f902443f626c6788ff0625a2183e155cd02bb13L14-R14))
*  Imported the ElementRef module and changed the type of the fileUpload property to ElementRef<HTMLInputElement> for better access to the file input element ([link](https://github.com/fylein/fyle-mobile-app/pull/2424/files?diff=unified&w=0#diff-87d7fb258f45220697c3e663d654e6157f4d6ee062e7c4ed5234b259f7c23bc8L1-R1), [link](https://github.com/fylein/fyle-mobile-app/pull/2424/files?diff=unified&w=0#diff-87d7fb258f45220697c3e663d654e6157f4d6ee062e7c4ed5234b259f7c23bc8L14-R14))
*  Refactored the getImageFromImagePicker method and extracted the logic for uploading a file into a separate uploadFileCallback method for improved modularity and readability ([link](https://github.com/fylein/fyle-mobile-app/pull/2424/files?diff=unified&w=0#diff-87d7fb258f45220697c3e663d654e6157f4d6ee062e7c4ed5234b259f7c23bc8L22-R64), [link](https://github.com/fylein/fyle-mobile-app/pull/2424/files?diff=unified&w=0#diff-87d7fb258f45220697c3e663d654e6157f4d6ee062e7c4ed5234b259f7c23bc8L61-R70))
*  Added the return types and access modifiers to the methods in the `camera-options-popup.component.ts` file for clarity and consistency ([link](https://github.com/fylein/fyle-mobile-app/pull/2424/files?diff=unified&w=0#diff-87d7fb258f45220697c3e663d654e6157f4d6ee062e7c4ed5234b259f7c23bc8L22-R64), [link](https://github.com/fylein/fyle-mobile-app/pull/2424/files?diff=unified&w=0#diff-87d7fb258f45220697c3e663d654e6157f4d6ee062e7c4ed5234b259f7c23bc8L61-R70))
*  Added a new file `camera-options-popup.component.spec.ts` that contains the unit tests for the CameraOptionsPopupComponent ([link](https://github.com/fylein/fyle-mobile-app/pull/2424/files?diff=unified&w=0#diff-984d4bd38acf3c2b32317ff28aad1560c98c6481e3c1199841b1472660f65ba0R1-R172))

## Clickup
https://app.clickup.com/t/85ztznvfe

## Code Coverage
`100% Statements 26/26`
`100% Branches 11/11`
`100% Functions 10/10`
`100% Lines 24/24`

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/add-edit-expense/camera-options-popup/camera-options-popup.component.ts [hunk 1]
```diff
@@ -32,35 +32,42 @@ export class CameraOptionsPopupComponent implements OnInit {
...
+      const dataUrl = await this.fileService.readFile(file);
+      this.popoverController.dismiss({
+        type: file.type,
+        dataUrl,
+        actionSource: 'gallery_upload',
+      });
+    } else {
+      this.closeClicked();
+
+      if (file?.size > MAX_FILE_SIZE) {
```
Comment (Dimple16) [inline src/app/fyle/add-edit-expense/camera-options-popup/camera-options-popup.component.ts:46]: why do we need this?
it'll anyway go inside due to the above else block na?

**Review by Dimple16 — APPROVED** [review]
one question?

File: src/app/fyle/add-edit-expense/camera-options-popup/camera-options-popup.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+    nativeElement.dispatchEvent(new Event('change'));
+
+    expect(component.onChangeCallback).toHaveBeenCalledOnceWith(nativeElement);
+    expect(nativeElement.click).toHaveBeenCalledTimes(1);
+    expect(trackingService.addAttachment).toHaveBeenCalledOnceWith({ Mode: 'Add Expense', Category: 'Camera' });
+  }));
+
+  describe('uploadFileCallback():', () => {
+    it('should read file if the size in less than maximum file size', fakeAsync(() => {
```
Comment (OmkarJ13) [inline src/app/fyle/add-edit-expense/camera-options-popup/camera-options-popup.component.spec.ts:129]: size is less than

File: src/app/fyle/add-edit-expense/camera-options-popup/camera-options-popup.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+      expect(fileService.readFile).toHaveBeenCalledOnceWith(file);
+      expect(popoverController.dismiss).toHaveBeenCalledOnceWith({
+        type: file.type,
+        dataUrl: 'file',
+        actionSource: 'gallery_upload',
+      });
+    }));
+
+    it('should dismiss modal and show size limit exceeded to the user', fakeAsync(() => {
```
Comment (OmkarJ13) [inline src/app/fyle/add-edit-expense/camera-options-popup/camera-options-popup.component.spec.ts:145]: this case is when the size is exceeds the max size right? if yes, let's mention this in the test description as well.

**Review by OmkarJ13 — APPROVED** [review]

File: src/app/fyle/add-edit-expense/camera-options-popup/camera-options-popup.component.ts (hunk 1 above)
Comment (Chethan-Fyle) [reply src/app/fyle/add-edit-expense/camera-options-popup/camera-options-popup.component.ts:46]: +1. Are we not showing error if the file size is equal to max allowed size?

**Review by Chethan-Fyle — APPROVED** [review]
One minor comment.

**Pull request merged**